From ea4cc18ae5a98869221285db834feb51c83fd1c9 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 23 Jun 2015 09:22:19 +0000
Subject: [PATCH 0592/1000] MANTIS 31677 Mantis 31677: ConveyorDev for CUSS 1.3
 Extend "ConveyorDev.idl" by - an additional moduleType "SAFETY_DOOR" - an
 additional errorCode "ERR_THERMAL_OVERLOAD"

git-svn-id: svn://localhost/SelfServiceCommon/trunk@866 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index be7ed339..07ace0a6 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.5";
+   const string interface_version = "1.6";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -74,7 +74,8 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
      BCR,                 // get barcodes of tags on baggage
      RFID_READER,         // get data from rfid-tags on baggage
      RFID_WRITER,         // set data to rfid-tags on baggage
-     IMAGE_SCANNER        // get image of baggage
+     IMAGE_SCANNER,       // get image of baggage
+     SAFETY_DOOR          // prevent unauthorized access
    };
 
    struct moduleCharacteristic
@@ -167,8 +168,9 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       ERR_EMERGENCY_STOP,           /**< conveyor emergency stop button pressed */
       ERR_USER_INTERFERENCE,        /**< user interference which are _not_ considered
                                          critical to health, safety and security */
-      ERR_SAFETY_INTRUSION          /**< safety intrusions which are considered
+      ERR_SAFETY_INTRUSION,         /**< safety intrusions which are considered
                                          critical to health, safety and security */
+      ERR_THERMAL_OVERLOAD          /**< too hot to work */
    };
 
    /**
@@ -187,8 +189,9 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
 
    struct moduleStatus
    {
-     moduleCharacteristic type;
-     subDeviceStatus status;
+      moduleCharacteristic type;    /**< Identify the particular device */
+      subDeviceStatus status;       /**< General device status */
+      string description;           /**< Human readable details */
    };
 
    typedef sequence<moduleStatus> moduleStatusList;
-- 
2.41.0.windows.1

